vlm/src/main/scala/geotrellis/contrib/vlm/spark/RasterSummary.scala

```diff
@@ -63,53 +63,63 @@ case class RasterSummary(
+  def toTileLayerMetadata(layoutLevel: LayoutLevel): (TileLayerMetadata[K], Some[Int]) = {
     val LayoutLevel(zoom, layoutDefinition) = layoutLevel
     // We want to align the data extent to pixel layout of layoutDefinition
     val layerGridExtent = layoutDefinition.createAlignedGridExtent(extent)
-    val dataBounds = KeyBounds(layoutDefinition.mapTransform.extentToBounds(layerGridExtent.extent))
-    val tlm = TileLayerMetadata[SpatialKey](cellType, layoutDefinition, layerGridExtent.extent, crs, dataBounds)
+    val keyBounds = bounds match {
+      case KeyBounds(minKey, maxKey) =>
+        val KeyBounds(minKeySpatial, maxKeySpatial) = KeyBounds(layoutDefinition.mapTransform.extentToBounds(layerGridExtent.extent))
+        KeyBounds(minKey.setComponent(minKeySpatial), maxKey.setComponent(maxKeySpatial))
+      case EmptyBounds => EmptyBounds
+    }
+    val tlm = TileLayerMetadata(cellType, layoutDefinition, layerGridExtent.extent, crs, keyBounds)
     (tlm, Some(zoom))
   }
 
-  def toTileLayerMetadata(layoutDefinition: LayoutDefinition, zoom: Int): (TileLayerMetadata[SpatialKey], Some[Int]) =
+  def toTileLayerMetadata(layoutDefinition: LayoutDefinition, zoom: Int): (TileLayerMetadata[K], Some[Int]) =
     toTileLayerMetadata(LayoutLevel(zoom, layoutDefinition))
 
-  def toTileLayerMetadata(layoutType: LayoutType): (TileLayerMetadata[SpatialKey], Option[Int]) = {
+  def toTileLayerMetadata(layoutType: LayoutType): (TileLayerMetadata[K], Option[Int]) = {
     val (ld, zoom) = layoutType.layoutDefinitionWithZoom(crs, extent, cellSize)
-    val dataBounds: Bounds[SpatialKey] = KeyBounds(ld.mapTransform.extentToBounds(extent))
-    val tlm = TileLayerMetadata[SpatialKey](cellType, ld, extent, crs, dataBounds)
+    val keyBounds = bounds match {
+      case KeyBounds(minKey, maxKey) =>
+        val KeyBounds(minKeySpatial, maxKeySpatial) = KeyBounds(ld.mapTransform.extentToBounds(extent))
+        KeyBounds(minKey.setComponent(minKeySpatial), maxKey.setComponent(maxKeySpatial))
+      case EmptyBounds => EmptyBounds
+    }
+    val tlm = TileLayerMetadata(cellType, ld, extent, crs, keyBounds)
     (tlm, zoom)
   }
 
   // TODO: probably this function should be removed in the future
-  def resample(resampleGrid: ResampleGrid[Long]): RasterSummary = {
+  def resample(resampleGrid: ResampleGrid[Long]): RasterSummary[K] = {
     val re = resampleGrid(toGridExtent)
     RasterSummary(
       crs      = crs,
       cellType = cellType,
       cellSize = re.cellSize,
       extent   = re.extent,
       cells    = re.size,
-      count    = count
+      count    = count,
+      bounds   = bounds // do nothing with it, since it contains non spatial information
     )
   }
 }
 
 object RasterSummary {
-  /** Collect [[RasterSummary]] from unstructred rasters, grouped by CRS */
-  def collect[V <: CellGrid[N]: GetComponent[?, ProjectedExtent], N: Integral](rdd: RDD[V]): Seq[RasterSummary] = {
```

Comment:
Later we can add some `RasterSourceMetadata` component to pull out some important information from anything that has a defined `Lens`, but we definitely don't need such a generic abstraction now.